core/include/detray/grids/grid2.hpp

```diff
@@ -229,6 +229,15 @@ class grid2 {
             _axis_p0, _axis_p1, _axis_p0.bin(p2[0]), _axis_p1.bin(p2[1]))];
     }
 
+    /// Stub function until the zone call is working correctly
+    template <typename detector_t, typename track_t>
+    DETRAY_HOST_DEVICE dindex_range
+    search(const detector_t & /*det*/,
```

Comment:
Yes, I will put a grid surface finder in between at some point. The grid itself will then only provide the sequence of surface indices in a neighborhood, but I removed that for now, because I think it does not work on device.